code/modules/projectiles/projectile/beams.dm

```diff
@@ -368,21 +368,22 @@
 	hitsound = null
 	damage = 0
 	damage_type = STAMINA
-	var/suit_types = list(/obj/item/clothing/suit/redtag, /obj/item/clothing/suit/bluetag)
 	impact_effect_type = /obj/effect/temp_visual/impact_effect/blue_laser
 	light_color = LIGHT_COLOR_BLUE
+	var/lasertag_team = "neutral"
 
 /obj/projectile/beam/lasertag/on_hit(atom/target, blocked = 0, pierce_hit)
 	. = ..()
 	if(ishuman(target))
 		var/mob/living/carbon/human/M = target
-		if(istype(M.wear_suit))
-			if(M.wear_suit.type in suit_types)
+		var/datum/component/lasertag/comp = M?.GetComponent(/datum/component/lasertag)
+		if(comp)
+			if(lasertag_team != comp.team_color)
```

Comment:
Lastly, here is where components can shine

We can have the component generically hook `COMSIG_ATOM_BULLET_ACT` to react to being hit by a bullet, check `istype(projectile, /obj/projectile/beam/lasertag)`, then apply the stamina damage